metrics/web/test/dashboard/presentation/state/project_metrics_notifier_test.dart

```diff
@@ -313,20 +314,23 @@ void main() {
     test(
         ".filterByProjectName() filters list of the project metrics according to the given value",
         () async {
+      final metricsTileViewModel =
+          projectMetricsNotifier.projectsMetricsTileViewModels.last;
+
       final expectedProjectMetrics = [
-        projectMetricsNotifier.projectsMetricsTileViewModels.last
+        metricsTileViewModel,
       ];
+      final projectNameFilter = metricsTileViewModel.projectName;
 
-      final projectNameFilter = expectedProjectMetrics.first.projectName;
-      final listener = expectAsync0(() {
-        final filteredProjectMetrics =
-            projectMetricsNotifier.projectsMetricsTileViewModels;
+      final listener = expectAsyncUntil0(
+        () {},
+        () {
+          final filteredProjectMetrics =
+              projectMetricsNotifier.projectsMetricsTileViewModels;
+          return listEquals(filteredProjectMetrics, expectedProjectMetrics);
+        },
+      );
 
-        expect(
-          filteredProjectMetrics,
-          equals(expectedProjectMetrics),
-        );
-      });
       projectMetricsNotifier.addListener(listener);
       projectMetricsNotifier.filterByProjectName(projectNameFilter);
     });
```

Comment:
I think we should add a comma here and format the code, to make the alignment look better.
```suggestion
    },);
```